### PR TITLE
Fix build with older gcc versions

### PR DESCRIPTION
**Pull request by MrAnno**
This is a fix for something that was introduced in #405, so no changelog entry is needed.

@Snaipe
The PR is opened against `bleeding`, but we are in the middle of a release. Which workflow do you prefer? Should I change the base to `master` and then merge that branch back into `bleeding`?

### CODE REVIEW

File: test/unit/asprintf.c
Comment (Snaipe) [inline test/unit/asprintf.c:20]: I'm surprised gcc even cares given that any value shorter than int gets implicitly passed as an int through varargs. I guess the warning is there just for correctness.

**Review by Snaipe — APPROVED** [review]
LGTM.

File: test/unit/asprintf.c
Comment (MrAnno) [reply test/unit/asprintf.c:20]: I was surprised too. :)

**Comment by Snaipe** [comment]
Is this OK to merge? Feel free to do so yourself if you're happy with your changes, since they've already got my approval and haven't since changed.

**Comment by MrAnno** [comment]
I think it is ready. Thanks for your review.

**Pull request merged**
